src/ucs/sys/topo/base/topo.c

```diff
@@ -383,8 +383,16 @@ ucs_topo_get_distance_sysfs(ucs_sys_device_t device1,
                             ucs_sys_device_t device2,
                             ucs_sys_dev_distance_t *distance)
 {
-    char path1[PATH_MAX], path2[PATH_MAX], common_path[PATH_MAX];
-    ucs_status_t status;
+    char *path1 = ucs_malloc(PATH_MAX, "path1");
+    char *path2 = ucs_malloc(PATH_MAX, "path2");
+    char *common_path = ucs_malloc(PATH_MAX, "common_path");
+    ucs_status_t status = UCS_OK;
+
+    if ((path1 == NULL) || (path2 == NULL) || (common_path == NULL)) {
+        ucs_error("Failed to allocate memory for path buffers");
+        status = UCS_ERR_NO_MEMORY;
+        goto out;
+    }
```

Comment:
That condition can lead to memory leaks (e.g. when only `common_path` allocation failed so we need to free two other pointers before calling return)

Reply:
fixed